### DISCUSSION TITLE
LoRaWAN in action :)

### DISCUSSION THREAD

**Discussion by StevenCellist** [jgromes/RadioLib#881] [Show and tell]
The LoRaWAN stack in action is clearly visible using [this](https://ttnmapper.org/devices/?device=ttn-mapper&startdate=2023-11-24&enddate=2023-11-24&gateways=on&lines=on&points=on) link.
I built a LoRaWAN Mapper using RadioLib and it's working fine just fine! Woohoo!
For anyone interested.. I am planning to sell these devices in a few weeks :)
![WhatsApp Image 2023-11-24 at 12 20 25_65297e1b](https://github.com/jgromes/RadioLib/assets/47155822/99d01f07-a8ef-4b16-a398-228c1695f75b)

**Comment by ksjh** [comment]
Hi @StevenCellist, this is a great implementation. I am currently using a similar solution, but with 3000mAh LiPo cells made by MakerFocus. Looking at the map, I see that your setup receives data from 15 GPS satellites most of the time. I am not sure if you are using the standard TinyGPS++ library, but this lib does just rely on GPS, no other constellations. The version the board manufacturer provides has been patched to include the messages `$GNGGA` and `$GNRMC` additionally to the GPS versions `$GPGGA` and `$GPRMC`. With another GNSS library like MicroNMEA and an external L1/L5 antenna, I am able to receive GNSS data from 20 satellites *indoors* and get position fixes much faster since it uses all constellations. You might need to send an initialization string `$CFGSYS,h35155*68` to enable the additional GNSS systems.

**Reply by StevenCellist** [reply]
Duuude you're coming in clutch here. I've been investigating the GPS performance the past couple of days.
The fact that the number of satellites on the map reports at a maximum of 15 is because I only reserved 4 bits for that field, since anything above 15 does not tell that much information; I can receive over 20 without problems.
I did indeed notice that the board outputs much more than the basic NMEA protocol, but have been focused on user experience the past weeks, since I recently got one of my devices out to an experienced tester. Since that's now mostly fixed, I started looking at this.
Is the integrated antenna capable of receiving these L1/L5 signals, or do you need the extra antenna for that? And can you please recommend me some antennas? I see a wild variety of sizes and prices after a quick search.

I've also been trying to activate the GPS backup power, but I can't really seem to get it working. Have you used that in any way or can you share any thoughts on it?

This is what we are a community for.. been working my ass of to get a good LoRaWAN stack into RadioLib and it's really awesome to receive this in return! You made my day :)

**Reply by ksjh** [reply]
Hi @StevenCellist, glad that the info was useful. Thanks for your excellent LoRaWAN implementation. Should we move the discussion elsewhere, since not all is related to RadioLib? So, just briefly:

- I think the integrated antenna is not really meant for L5. The usual L1/L5 patch antennas look differently. You might be able to receive really strong L5 signals occasionally.
- You can try to find the UPrecise application on the web and program the board to output all GNSS data via USB CDC so that you can visualize the GNSS NEMA sentences. Doing so will show signal reception for different constellations and frequencies.
- I used a cheap external antenna bought at AliExpress, no-name, active, with SMA connector and an U.Fl to SMA pigtail. I can send you the link somehow, but I do not want to spam the discussion here.
- Did you try to connect something (e.g., a button cell) to test pad T3 (Vbat) and T4 (GND)? Looking at the schematic, a battery connected to the main battery connector of the board should feed the backup supply input of the GNSS chip anyway, even when `Vext` is disabled. So, unless your main battery is disconnected or completely flat, you would not notice any difference. I did not try an additional button cell.

**Reply by StevenCellist** [reply]
Are you on the Heltec forum? Would be a good place for this discussion since we're using their board. Would be glad to discuss this more! My GPS knowledge is very limited - just about enough understanding of the basics to do some mapping, very happy to learn more.

**Reply by ksjh** [reply]
It took some time to get my account activated there, but now I am also on the Heltec forum, same user name as here.

**Reply by StevenCellist** [reply]
Continuing on the [Heltec forum](http://community.heltec.cn/t/wireless-tracker-gnss-performance/14428)...
